client-sdk/go/modules/contracts/types.go

```diff
@@ -179,6 +180,70 @@ type InstanceStorageQueryResult struct {
 	Value []byte `json:"value"`
 }
 
+// StoreKind defines the public or confidential store type for performing queries.
+type StoreKind uint32
+
+// MarshalText returns string representation of StoreKind.
+func (sk StoreKind) MarshalText() string {
```

Comment:
This signature is wrong, please check [`encoding.TextMarshaller`](https://pkg.go.dev/encoding#TextMarshaler).